File: Quick Starts/ExceptionHandling/CS/ExceptionHandlingBasicQuickStart/AppMessageExceptionHandler.cs
//===============================================================================
// Microsoft patterns & practices Enterprise Library
// Exception Handling Application Block QuickStart
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Microsoft.Practices.EnterpriseLibrary.Common.Configuration;
using Microsoft.Practices.EnterpriseLibrary.ExceptionHandling;
using Microsoft.Practices.EnterpriseLibrary.ExceptionHandling.Configuration;
using Microsoft.Practices.EnterpriseLibrary.Common.Configuration.ObjectBuilder;
using System.Collections.Specialized;

namespace ExceptionHandlingQuickStart
{
	/// <summary>
	/// Summary description for GlobalPolicyExceptionHandler.
	/// </summary>
    [ConfigurationElementType(typeof(CustomHandlerData))]
    public class AppMessageExceptionHandler : IExceptionHandler
	{
		public AppMessageExceptionHandler(NameValueCollection ignore)
		{
		}

        public Exception HandleException(Exception exception, Guid handlingInstanceId)
        {
            DialogResult result = this.ShowThreadExceptionDialog(exception);

            return exception;
        }

		// Creates the error message and displays it.
		private DialogResult ShowThreadExceptionDialog(Exception ex) 
		{
			string errorMsg = "The following exception was caught by the Quick Start Global Exception Handler:" + Environment.NewLine + Environment.NewLine;

			StringBuilder sb = new StringBuilder();
			StringWriter writer = new StringWriter(sb);

			AppTextExceptionFormatter formatter = new AppTextExceptionFormatter(writer, ex);
			
			// Format the exception
			formatter.Format();

			errorMsg +=  sb.ToString();

			return MessageBox.Show(errorMsg, "Application Error", MessageBoxButtons.OK, MessageBoxIcon.Stop);
		}
	    
	}
}

File: Quick Starts/ExceptionHandling/VB/ExceptionHandlingBasicQuickStart/AppMessageExceptionHandler.vb
'===============================================================================
' Microsoft patterns & practices Enterprise Library
' Exception Handling Application Block QuickStart
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

<ConfigurationElementType(GetType(CustomHandlerData))> _
    Public Class AppMessageExceptionHandler
    Implements IExceptionHandler

    Public Sub New(ByVal ignore As NameValueCollection)

    End Sub


    Public Function HandleException(ByVal e As Exception, ByVal handlingInstanceID As Guid) As Exception Implements IExceptionHandler.HandleException

        Dim result As DialogResult = Me.ShowThreadExceptionDialog(e)

        Return e
    End Function

    ' Creates the error message and displays it.
    Private Function ShowThreadExceptionDialog(ByVal e As Exception) As DialogResult

        Dim errorMsg As String = "The following exception was caught by the Quick Start Global Exception Handler:" + Environment.NewLine + Environment.NewLine

        Dim sb As StringBuilder = New StringBuilder
        Dim writer As StringWriter = New StringWriter(sb)

        Dim formatter As AppTextExceptionFormatter = New AppTextExceptionFormatter(writer, e)

        ' Format the exception
        formatter.Format()

        errorMsg += sb.ToString()

        Return MessageBox.Show(errorMsg, "Application Error", MessageBoxButtons.OK, MessageBoxIcon.Stop)
    End Function

End Class

